Code formatting: Introduce black

He3lixxx opened the pull request:
Continuation of #1585 

Todos left:
* [x] Discuss whether we actually want this
* [x] Update to current version
* [x] ~Exclude settings.py?~
* [x] check for formatting fails (I will just re-check the ones given in #1585)
* [x] Fail CI if not formatted properly
* [x] Check for conflicts with pylint
* [x] enable string normalization
* [x] Management command (for black and pylint)
* [x] create .git-blame-ignore-revs (will do last, when commit hash is fixed)

Nice to haves:
* [ ] Pyproject.toml: Move pylint settings there? Reference: https://github.com/ephios-dev/ephios/blob/main/pyproject.toml
* [ ] Add isort

janno42 commented on evap/evaluation/models.py:
```diff
@@ -848,64 +957,87 @@ def annotate_with_participant_and_voter_counts(cls, evaluation_query):
+        unique_together = (("evaluation", "contributor"),)
+        ordering = [
+            "order",
+        ]
```
this feels inconsistent, but i don't think it's important to change it

janno42 reviewed (approved):
i looked trough the tiny diff and think its ready to :shipit:

He3lixxx commented:
I've added a .git-ignore-blame-rev and updated https://github.com/e-valuation/EvaP/wiki/Development-Tips with instructions on how to use it.

For new contributors, it might make sense to put a note in the README.md that they probably want their IDE to lint + format files on save or something? Since there might be discussion about this, I'll make a new PR for that. Pyproject.toml merge will also go in there.
